include/deal.II/base/symmetric_tensor.h

```diff
@@ -1068,6 +1084,289 @@ namespace internal
+        //
+        // where mult is a 3x3 matrix with
+        // entries [[1,0,0],[0,1,0],[0,0,2]] to
+        // capture the fact that we need to add up
+        // both the c_ij12*a_12 and the c_ij21*a_21
+        // terms
+        //
+        // in addition, in this scheme, the
+        // identity tensor has the matrix
+        // representation mult^-1.
+        //
+        // the inverse of 't' therefore has the
+        // matrix representation
+        //
+        // inv.data = mult^-1 * t.data^-1 * mult^-1
+        //
+        // in order to compute it, let's first
+        // compute the inverse of t.data and put it
+        // into tmp.data; at the end of the
+        // function we then scale the last row and
+        // column of the inverse by 1/2,
+        // corresponding to the left and right
+        // multiplication with mult^-1
+        const Number t4 = t.data[0][0]*t.data[1][1],
+                     t6 = t.data[0][0]*t.data[1][2],
+                     t8 = t.data[0][1]*t.data[1][0],
+                     t00 = t.data[0][2]*t.data[1][0],
+                     t01 = t.data[0][1]*t.data[2][0],
+                     t04 = t.data[0][2]*t.data[2][0],
+                     t07 = 1.0/(t4*t.data[2][2]-t6*t.data[2][1]-
+                                t8*t.data[2][2]+t00*t.data[2][1]+
+                                t01*t.data[1][2]-t04*t.data[1][1]);
+        tmp.data[0][0] = (t.data[1][1]*t.data[2][2]-t.data[1][2]*t.data[2][1])*t07;
+        tmp.data[0][1] = -(t.data[0][1]*t.data[2][2]-t.data[0][2]*t.data[2][1])*t07;
+        tmp.data[0][2] = -(-t.data[0][1]*t.data[1][2]+t.data[0][2]*t.data[1][1])*t07;
+        tmp.data[1][0] = -(t.data[1][0]*t.data[2][2]-t.data[1][2]*t.data[2][0])*t07;
+        tmp.data[1][1] = (t.data[0][0]*t.data[2][2]-t04)*t07;
+        tmp.data[1][2] = -(t6-t00)*t07;
+        tmp.data[2][0] = -(-t.data[1][0]*t.data[2][1]+t.data[1][1]*t.data[2][0])*t07;
+        tmp.data[2][1] = -(t.data[0][0]*t.data[2][1]-t01)*t07;
+        tmp.data[2][2] = (t4-t8)*t07;
+
+        // scale last row and column as mentioned
+        // above
+        tmp.data[2][0] /= 2;
+        tmp.data[2][1] /= 2;
+        tmp.data[0][2] /= 2;
+        tmp.data[1][2] /= 2;
+        tmp.data[2][2] /= 4;
+
+        return tmp;
+      }
+    };
+
+
+    template<typename Number>
+    struct Inverse<4,3,Number>
+    {
+      static dealii::SymmetricTensor<4,3,Number>
+      value (const dealii::SymmetricTensor<4,3,Number> &t)
+      {
+        dealii::SymmetricTensor<4,3,Number> tmp = t;
+
+        // this function follows the exact same
+        // scheme as the 2d case, except that
+        // hardcoding the inverse of a 6x6 matrix
+        // is pretty wasteful. instead, we use the
+        // Gauss-Jordan algorithm implemented for
+        // FullMatrix; the following code is copied
+        // from there because using the FullMatrix
+        // class would introduce circular
+        // references between libbase and liblac
```

Comment:
Yeah, I'm just going to remove the comment but leave the code as is for now. Trying to make sure that FullMatrix works with all number types opens up a whole new realm of potential issues. Besides, one would also have to use two copy operations which would be wasteful.